lib/token.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const BaseModel = require('./base');
+const nodeify = require('./nodeify');
+const iron = require('iron');
+
+// Symbols for private members
+const password = Symbol('password');
+
+class TokenModel extends BaseModel {
+    /**
+     * Construct a TokenModel object
+     * @method constructor
+     * @param  {Object}    config
+     * @param  {Object}    config.datastore         Object that will perform operations on the datastore
+     * @param  {Number}    config.userId            The ID of the associated user
+     * @param  {String}    config.value             The token value
+     * @param  {String}    config.tokenId           The token ID
+     * @param  {String}    config.name              The token name
+     * @param  {String}    config.description       The token description
+     * @param  {Date}      config.lastUsed          The last time the token was accessed
+     * @param  {String}    config.password          The encryption password
+     */
+    constructor(config) {
+        super('token', config);
+        this[password] = config.password;
+    }
+
+    /**
+     * Update a token with sealed token value
+     * @method update
+     * @return {Promise}
+     */
+    update() {
```

Comment:
My understanding is that any time we go to store something like a token value, we want it to be encrypted. That happens either when creating or updating a model.

The factory is what handles creating new models (and getting existing ones), whereas once you have a model you call update() on the model itself, which means both of them need to be able to encrypt the token values. @d2lam can you confirm and/or elaborate please?